app/assets/javascripts/modules/timeline/timeline.coffee

```diff
@@ -456,13 +456,20 @@ do (document, ko, $=jQuery, config=@edsc.config, plugin=@edsc.util.plugin, strin
       Math.floor((p - originPx) * scale + start)
 
     zoomIn: ->
+      @_centerTemporalSelection()
       @root.trigger('buttonzoom')
       @_deltaZoom(-1)
 
     zoomOut: ->
+      @_centerTemporalSelection()
       @root.trigger('buttonzoom')
       @_deltaZoom(1)
 
+    _centerTemporalSelection: =>
```

Comment:
Also needs to center on the focused time span if the user has just clicked on a date.